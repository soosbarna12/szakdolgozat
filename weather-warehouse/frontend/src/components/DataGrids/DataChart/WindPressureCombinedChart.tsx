import { Skeleton } from "@mui/material";
import React from "react";
import { Bar, CartesianGrid, ComposedChart, Legend, Line, ResponsiveContainer, Tooltip, XAxis, YAxis } from "recharts";
import { HistoricalDataTable } from "../../../types/historicalDataTable.type";

export function WindPressureCombinedChart({ data }: { data: HistoricalDataTable[] }) {
  if (!data || data.length === 0) {
    return (
      <Skeleton
        variant="rectangular"
        animation="wave"
        width="100%"
        height="100%"
        data-testid="windPressureSkeleton"
        sx={{ borderRadius: '10px' }}
      />
    )
  }

  return (
    <ResponsiveContainer width="100%" height="100%">
      <ComposedChart data={data}>
        <CartesianGrid strokeDasharray="3 3" />
        <XAxis dataKey="date" />
        <YAxis yAxisId="left" />
        <YAxis yAxisId="right" orientation="right" />
        <Tooltip />
        <Legend />
        <Bar yAxisId="left" dataKey="pressure" fill="#8884d8" name="Pressure (hPa)" />
        <Line yAxisId="right" type="monotone" dataKey="windSpeed" stroke="#82ca9d" name="Wind Speed (m/s)" />
      </ComposedChart>
    </ResponsiveContainer>
  );
}