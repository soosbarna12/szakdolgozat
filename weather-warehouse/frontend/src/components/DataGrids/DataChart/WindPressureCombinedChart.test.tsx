import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import React from 'react';
import { renderWithQueryClient } from '../../../utils/test/renderWithQueryClient';
import { WindPressureCombinedChart } from './WindPressureCombinedChart';

// eslint-disable-next-line @typescript-eslint/no-require-imports
global.ResizeObserver = require("resize-observer-polyfill");

jest.mock("recharts", () => ({
    ResponsiveContainer: ({ children }: { children: React.ReactNode }) => (
        <div data-testid="windPressureChartContainer">{children}</div>
    ),
    ComposedChart: ({ children }: { children: React.ReactNode }) => (
        <div data-testid="windPressureChart">{children}</div>
    ),
    CartesianGrid: () => <div data-testid="mockCartesianGrid" />,
    XAxis: () => <div data-testid="mockXAxis" />,
    YAxis: () => <div data-testid="mockYAxis" />,
    Tooltip: () => <div data-testid="mockTooltip" />,
    Legend: () => <div data-testid="mockLegend" />,
    Bar: () => <div data-testid="mockBar">Mock Bar</div>,
    Line: () => <div data-testid="mockLine">Mock Line</div>,
}));

describe('DataGrids/WindPressureCombinedChart', () => {
    const mockData = [
        { date: '2023-01-01', pressure: 1015, windSpeed: 5 },
        { date: '2023-01-02', pressure: 1020, windSpeed: 8 },
    ];

    beforeAll(() => {
        global.ResizeObserver = class {
            observe() { }
            unobserve() { }
            disconnect() { }
        };
    });

    const renderComponent = (data = mockData) => {
        return renderWithQueryClient(<WindPressureCombinedChart data={data as any} />);
    };

    it('matches WindPressureCombinedChart component snapshot', () => {
        const component = renderComponent();
        expect(component).toMatchSnapshot();
    });

    it('renders skeleton when data is empty', () => {
        renderComponent([]);
        const skeleton = screen.getByTestId('windPressureSkeleton');
        expect(skeleton).toBeInTheDocument();
    });

    it('renders chart with data', () => {
        renderComponent();
        const chartContainer = screen.getByTestId('windPressureChartContainer');
        const chart = screen.getByTestId('windPressureChart');

        expect(chartContainer).toBeInTheDocument();
        expect(chart).toBeInTheDocument();
    });
});