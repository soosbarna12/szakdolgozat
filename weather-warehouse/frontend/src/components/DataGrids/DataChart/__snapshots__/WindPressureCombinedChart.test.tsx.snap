// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataGrids/WindPressureCombinedChart matches WindPressureCombinedChart component snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="windPressureChartContainer"
      >
        <div
          data-testid="windPressureChart"
        >
          <div
            data-testid="mockCartesianGrid"
          />
          <div
            data-testid="mockXAxis"
          />
          <div
            data-testid="mockYAxis"
          />
          <div
            data-testid="mockYAxis"
          />
          <div
            data-testid="mockTooltip"
          />
          <div
            data-testid="mockLegend"
          />
          <div
            data-testid="mockBar"
          >
            Mock Bar
          </div>
          <div
            data-testid="mockLine"
          >
            Mock Line
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="windPressureChartContainer"
    >
      <div
        data-testid="windPressureChart"
      >
        <div
          data-testid="mockCartesianGrid"
        />
        <div
          data-testid="mockXAxis"
        />
        <div
          data-testid="mockYAxis"
        />
        <div
          data-testid="mockYAxis"
        />
        <div
          data-testid="mockTooltip"
        />
        <div
          data-testid="mockLegend"
        />
        <div
          data-testid="mockBar"
        >
          Mock Bar
        </div>
        <div
          data-testid="mockLine"
        >
          Mock Line
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
